neurokit2/complexity/fractal_dfa.py

```diff
@@ -291,6 +291,132 @@ def fractal_dfa(
+    # Turn to list
+    if isinstance(q, str):
+        if multifractal is False:
+            q = [2]
+        else:
+            q = [-5, -3, -1, 0, 1, 3, 5]
+    elif isinstance(q, (int, float)):
+        q = [q]
+
+    # Fractal powers as floats
+    q = np.asarray_chkfinite(q, dtype=float)
+
+    return q
+
+
+def _slopes(scale, fluctuations, q):
+    # Extract the slopes of each `q` power obtained with MFDFA to later produce
+    # either the singularity spectrum or the multifractal exponents.
+    # Note: added by Leonardo Rydin (see https://github.com/LRydin/MFDFA/)
+
+    # Ensure mfdfa has the same q-power entries as q
+    if fluctuations.shape[1] != q.shape[0]:
+        raise ValueError("Fluctuation function and q powers don't match in dimension.")
+
+    # Allocated array for slopes
+    slopes = np.zeros(len(q))
+    # Find slopes of each q-power
+    for i in range(len(q)):
+        # if fluctiations is zero, log2 wil encounter zero division
+        old_setting = np.seterr(divide="ignore", invalid="ignore")
+        slopes[i] = np.polyfit(np.log2(scale), np.log2(fluctuations[:, i]), 1)[0]
+        np.seterr(**old_setting)
+
+    return slopes
+
+
+def _fractal_dfa_getwindow(signal, n, window, overlap=True):
+    # This function reshapes the segments from a one-dimensional array to a
+    # matrix for faster polynomail fittings. 'Overlap' reshapes into several
+    # overlapping partitions of the data
+
+    # TODO: see whether this step could be integrated within complexity_coarsegraining
+
+    if overlap:
+        segments = np.array([signal[i : i + window] for i in np.arange(0, n - window, window // 2)])
+    else:
+        segments = signal[: n - (n % window)]
+        segments = segments.reshape((signal.shape[0] // window, window))
+
+    return segments
+
+
+def _fractal_dfa_trends(segments, window, order=1):
+    # TODO: can we rely on signal_detrend?
+    x = np.arange(window)
+
+    coefs = np.polyfit(x[:window], segments.T, order).T
+
+    # TODO: Could this be optimized? Something like np.polyval(x[:window], coefs)
+    trends = np.array([np.polyval(coefs[j], x) for j in np.arange(len(segments))])
+
+    return trends
+
+
+def _fractal_dfa_fluctuation(segments, trends, q=2):
+
+    # Detrend
+    detrended = segments - trends
+
+    # Compute variance
+    var = np.var(detrended, axis=1)
+
+    # Find where q is close to zero. Limit set at |q| < 0.1
+    # See https://github.com/LRydin/MFDFA/issues/33
+    is0 = np.abs(q) < 0.1
+
+    # Reshape q to perform np.float_power
+    q_non0 = q[~is0].reshape(-1, 1)
+
+    # Get the fluctuation function, which is a function of the windows and of q
+    # When q = 2 (i.e., multifractal = False)
+    # The formula is equivalent to np.sqrt(np.mean(var))
+    # And corresponds to the Root Mean Square (RMS)
+    fluctuation = np.float_power(np.mean(np.float_power(var, q_non0 / 2), axis=1), 1 / q_non0.T)
+
+    if np.sum(is0) > 0:
+        fluc0 = np.exp(np.mean(np.log(var)))
```

Comment:
what I did is that I computed the values for q=0 here and then inserted them in the fluctuation list

Reply:
Could you maybe give this a try (sorry for not trying myself, I have to finish something today, I was just puzzled with the disparity in the plot. I think it has to do with a `0.5` that is missing):

```python
fluc0 = np.exp(0.5*np.mean(np.log(var)))
```

Reply:
![image](https://user-images.githubusercontent.com/8875533/181870356-41b26c6c-7bb8-4895-b772-309120aa3679.png)

ah yes that looks better!